custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -246,6 +253,35 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
         locationsService.onSelectLocation($item, level, locationsCache, vm);
     };
 
+    vm.onSelect = function($item, level) {
```

Comment:
I think the place where this is called was meant to be a recursive call in line 245 of the location service, and that should also be fixed there, so other pages using the service have the correct behavior.

Is this just copied back from the old `onSelect`, if so, it should be renamed `onSelectLocation` and can entirely replace the one two lines above